#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
}
Viewpoint {
  orientation 1 0 0 4.71238898038469
  position -0.20005604321626272 2.929374456934362 0.17264475627195797
}
TexturedBackground {
}
TexturedBackgroundLight {
  castShadows FALSE
}
DEF CITY Solid {
  rotation 0 1 0 3.14159
  children [
    Floor {
      size 3 3
      appearance SquarePavement {
      }
    }
    DEF TWO_WALL Solid {
      translation 0.5 0.15 -0.47
      rotation 0 1 0 1.5708
      children [
        DEF TWO_WALL_SHAPE Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/old_brick_wall.jpg"
              ]
            }
            roughness 1
            metalness 0
            textureTransform TextureTransform {
              scale 4 1
            }
          }
          geometry Box {
            size 1 0.3 0.02
          }
        }
      ]
      name "solid(6)"
      boundingObject USE TWO_WALL_SHAPE
    }
    DEF ONE_WALL Solid {
      translation -0.27 0.15 -0.47
      rotation 0 1 0 -3.1415853071795863
      children [
        DEF ONE_WALL Shape {
          appearance PBRAppearance {
            baseColorMap ImageTexture {
              url [
                "textures/old_brick_wall.jpg"
              ]
            }
            roughness 1
            metalness 0
            textureTransform TextureTransform {
              scale 2 1
            }
          }
          geometry Box {
            size 0.5 0.3 0.02
          }
        }
      ]
      name "solid(7)"
      boundingObject USE ONE_WALL
    }
    DEF WALLS Group {
      children [
        DEF TWOS Group {
          children [
            Solid {
              translation -0.00999994 0.15 0.03
              rotation 0 1 0 1.5708
              children [
                USE TWO_WALL_SHAPE
              ]
              name "solid(20)"
              boundingObject USE TWO_WALL_SHAPE
            }
            Solid {
              translation -0.01 0.15 -0.98
              children [
                USE TWO_WALL_SHAPE
              ]
              name "solid(22)"
              boundingObject USE TWO_WALL_SHAPE
            }
            Solid {
              translation 0.5 0.15 0.53
              children [
                USE TWO_WALL_SHAPE
              ]
              name "solid(25)"
              boundingObject USE TWO_WALL_SHAPE
            }
          ]
        }
        DEF ONES Group {
          children [
            Solid {
              translation 0.75 0.15 0.03
              children [
                USE ONE_WALL
              ]
              name "solid(31)"
              boundingObject USE ONE_WALL
            }
          ]
        }
      ]
    }
  ]
  name "solid(42)"
}
Robot {
  translation -1.03 0.04 -0.320001
  rotation 0 1 0 1.5708
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 0.917647 0.145098 0.145098
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    DEF WHEEL1 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.06 0 0.05
        rotation 1 0 0 0
        children [
          DEF WHEEL Transform {
            rotation 0 0 1 1.5708
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 0.305882 0.898039 0.25098
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.02
                  radius 0.04
                  subdivision 24
                }
              }
            ]
          }
        ]
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL2 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 0 0.05
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL3 HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.06 0 -0.05
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF WHEEL4 HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 0 -0.05
        rotation 1 0 0 0
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics Physics {
        }
      }
    }
    DEF DS_RIGHT DistanceSensor {
      translation -0.05 0 0.1
      rotation 0 1 0 3.14
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DEF DS_LEFT DistanceSensor {
      translation 0.05 0 0.1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    DEF DS_MID DistanceSensor {
      translation 0 0 0.1
      rotation 0 1 0 -1.5707953071795862
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.184314 0.596078 0.847059
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_mid"
      lookupTable [
        0 0 0
        0.5 5000 0
      ]
    }
    Compass {
    }
  ]
  boundingObject USE BODY
  physics Physics {
    density -1
    mass 1
  }
  controller "c_plus"
}
